docs/examples/MondrianTreeRegressor.md

```diff
@@ -0,0 +1,273 @@
+
+1. The upper and lower bounds of all the features in that particular node.
+2. The time of split, which is drawn from an exponential with mean \(\sum_{f=1}^D(u_b[f] - l_b[f])\)
+
+If you stare at the equation for half a minute, that makes sense as well because the larger the bounding box of a node, the time of split is smaller. Also the above property sets the time of split of leaves to infinity. Once the split feature and threshold are decided, then the tree construction happens in a similar way to that of a decision tree, a node splits the training data into two parts, \(X[:, f] < \delta\) to one leaf and \(X[:, f] > \delta\) to the other.
+
+###  Prediction time
+
+Recall that for a decision tree, for a new training point, computing the predictive mean and variance is fairly straightforward. That is, find the leaf node that a new training point lands in and output the mean and variance of the node.
+
+The prediction step of a Mondrian Tree is a bit more complicated. It takes into account all the nodes in the path of a new point from the root to the leaf for making a prediction. Mathematically, the distribution of
+\(P(Y | X)\) is given by
+
+\(P(Y | X) = \sum_{j} w_j \mathcal{N} (m_j, v_j)\)
+
+where the summation is across all the nodes in the path from the root to the leaf.
+
+To compute the weights given to each node, we need to compute the probability of separation \(p_j(x)\) of each node. This is computed in the following way.
+
+1. \(\Delta_{j} = \tau_{j} - \tau_{parent(j)}\)
+2. \(\eta_{j}(x) = \sum_{f}(\max(x[f] - u_{bj}[f], 0) + \max(0, l_{bj}[f] - x[f]))\)
+3. \(p_j(x) = 1 - e^{-\Delta_{j} \eta_{j}(x))}\)
+
+And then the weights are given by:
+
+If `j` is not a leaf:
+
+\(w_j(x) = p_j(x) \prod_{k \in anc(j)} (1 - p_k(x))\)
+
+If j is a leaf:
+
+\(w_j(x) = 1 - \sum_{k \in anc(j)} w_k(x)\)
+
+Let us take a more than half a minute to stare at these equations and understand what they mean.
+
+1. \(p_j(x)\) is high when \(\eta_{j}(x)\) is high. As \(\eta_{j}(x)\) approaches infinity, \(p_j(x)\) approaches zero. This means that when the point is far away from the bounding box of the node, the probability of separation becomes high.
+2. \(p_j(x)\) is high when \(\Delta{j}\) is high. This means when the bounding box of a node is small as compared to the bounding box of its parent, the probability of separation becomes high.
+3. The weight given to each node, \(w_j\) can be decomposed into two factors, \(p_j\) and \(\prod_{k \in anc(j)} (1 - p_k(x))\). The product term can be understood as the probability of that point not being separated till that particular node is reached. So the node where a new point just starts to branch of is given the highest weight.
+4. For a point in the training data, \(p_j(x)\) becomes zero all nodes (and hence \(w_j(x)\)). The leaf then has a weightage of 1.0 and this reduces to a standard decision tree prediction.
+
```

Comment:
We should also explain what happens for test points far away from the training data.